test/integration/smoke/test_baremetal.py

```diff
@@ -0,0 +1,155 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+""" Test for baremetal
+"""
+#Import Local Modules
+from marvin.cloudstackTestCase import cloudstackTestCase
+from marvin.lib.utils import cleanup_resources, validateList
+from marvin.lib.base import (Template,
+                             Host,
+                             ServiceOffering,
+                             Account,
+                             VirtualMachine)
+from marvin.lib.common import get_domain, get_zone
+from marvin.codes import PASS, HOSTTAG1
+from nose.plugins.attrib import attr
+
+#Import System modules
+_multiprocess_shared_ = True
+
+class TestBaremetal(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        testClient = super(TestBaremetal, cls).getClsTestClient()
+        cls.apiclient = testClient.getApiClient()
+        cls.services = testClient.getParsedTestDataConfig()
+        # Get Zone, Domain
+        cls.domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, testClient.getZoneForTests())
+        cls.services['mode'] = cls.zone.networktype
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+        cls.account = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            domainid=cls.domain.id
+        )
+        #Register baremetal template
+        cls.template = Template.register(
+            cls.apiclient,
+            cls.services["baremetal_template"],
+            zoneid=cls.zone.id,
+            account=cls.account.name,
+            domainid=cls.account.domainid,
+        )
+        cls.debug(
+            "Registered a template of format: %s with ID: %s" % (
+                cls.services["baremetal_template"]["format"], cls.template.id
+            )
+        )
+        #Update host with host tag
+        cls.hosts = Host.list(
+            cls.apiclient,
+            zoneid=cls.zone.id,
+            hypervisor=cls.hypervisor
+        )
+        if not validateList(cls.hosts)[0]:
+            assert False, "Invalid response for list host"
+        host_res = Host.update(
```

Comment:
I have checked this scenario, vm deployment is getting succeeded but we need to fix the baremetal planner a bit. So here no need to change anything.